### PR TITLE
SNOW-984450: Set ESLint rules to error

### PR DESCRIPTION
**Pull request by sfc-gh-pbulawa**
### Description
Set eslint rules to error.

### Checklist
- [x] Format code according to the existing code style (run `npm run lint:check -- CHANGED_FILES` and fix problems in changed code)
- [x] Create tests which fail without the change (if possible)
- [x] Make all tests (unit and integration) pass (`npm run test:unit` and `npm run test:integration`)
- [x] Extend the README / documentation and ensure is properly displayed (if necessary)
- [x] Provide JIRA issue id (if possible) or GitHub issue id in commit message

### CODE REVIEW

File: lib/file_transfer_agent/encrypt_util.js [hunk 1]
```diff
@@ -226,7 +227,7 @@ function EncryptUtil(encrypt, filestream, temp) {
     await new Promise((resolve) => {
```
Comment (sfc-gh-pmotacki) [inline lib/file_transfer_agent/encrypt_util.js:226]: It could be like new Promise((resolve, reject) => ...) without require('async')

**Review by sfc-gh-pmotacki — CHANGES_REQUESTED** [review]
Please verify one more time if any of changes isn't like a BCR for customers. I know that it isn't clear in js what is public and what is used by customer so it's only to be convinced that in our opinion we didn't change something look like public.

File: lib/file_transfer_agent/file_transfer_agent.js [hunk 2]
```diff
@@ -837,4 +837,7 @@ function FileTransferAgent(context) {
   }
 }
 
+//TODO: Create a function to renew expired client
```
[the remark this answers is not in the document]
Comment (sfc-gh-pbulawa) [reply lib/file_transfer_agent/file_transfer_agent.js]: Done

File: lib/file_transfer_agent/encrypt_util.js (hunk 1 above)
Comment (sfc-gh-pbulawa) [reply lib/file_transfer_agent/encrypt_util.js]: Done

**Review by sfc-gh-pmotacki — APPROVED** [review]
LGTM

**Pull request merged**